workbench-core/accounts/src/services/costCenterService.ts

```diff
@@ -34,6 +36,67 @@ export default class CostCenterService {
     this._dynamoDbService = dynamoDbService;
   }
 
+  /**
+   * Soft Delete Cost Center
+   * @param request - request for deleting cost center
+   * @returns void
+   */
+  public async softDeleteCostCenter(request: DeleteCostCenterRequest): Promise<void> {
+    const costCenterHaveProjects = await this._doesCostCenterHaveProjects(request.id);
+    if (costCenterHaveProjects) {
+      throw Boom.conflict(
+        `CostCenter ${request.id} cannot be deleted because it has project(s) associated with it`
+      );
+    }
+
+    const getCostCenterResponse = (await this._dynamoDbService
+      .get(buildDynamoDBPkSk(request.id, resourceTypeToKey.costCenter))
+      .execute()) as GetItemCommandOutput;
+
+    if (getCostCenterResponse.Item === undefined) {
+      throw Boom.notFound(`Could not find cost center ${request.id}`);
+    }
+
+    try {
+      await this._dynamoDbService
+        .update(buildDynamoDBPkSk(request.id, resourceTypeToKey.costCenter), {
```

Comment:
Your use of update will break when you merge in `develop` since I took out the `unmarshall` from `Updater.execute()`. When you need to fix this, just replace usage of `update(....).execute()` with `updateExecuteAndFormat(...)`. See https://github.com/aws-solutions/solution-spark-on-aws/pull/635 for examples of the refactor.

Reply:
This is good to know. Let me add a TODO here.